File: Poo_java/tp1/tp1_ex1/src/Point.java
import static java.lang.Math.sqrt;

public class Point {
    private float x ,y;

    public Point(float x , float y){
        this.x=x;
        this.y=y;

    }
    public Point(){
        this.x=0;
        this.y=0;
    }
    public void affiche (){
        System.out.println("x = " +x);
        System.out.println("y = " +y);
    }

    public void deplace(float dx , float dy){
        this.x=this.x+dx;
        this.y=this.y+dy;
    }



}

File: TpPoo_cpp/tp1_ex1/point.cpp
//
// Created by amani on 16/02/24.
//

#include <iostream>
#include "point.h"

point::point(float x,float y){
        abs=x;
        ord=y;
    }
    void point:: deplace(int dx, int dy){
        abs=abs+dx;
        ord=ord+dy;
    }
    void point :: affiche(){
        std::cout<<"x = "<<abs;
        std::cout<<"\ny = "<<ord;
    }

int main() {

    std::cout<<" ********** Ex1 ************ \n";
    point p (5,7);
    std::cout<<" before calling  deplace function \n";
    p.affiche();
    p.deplace(5,2);
    std::cout<<" \nafter calling  deplace function \n";
    p.affiche();
    return 0;
}





